target_s3_csv/__init__.py

```diff
@@ -151,6 +151,11 @@ def persist_messages(messages, config, s3_client):
             schema = utils.float_to_decimal(o['schema'])
             validators[stream] = Draft7Validator(schema, format_checker=FormatChecker())
             key_properties[stream] = o['key_properties']
+            
+            filename = o['stream'] + '-schema.json'
+            with open(filename, 'w+') as f:
+                f.write(o['stream'])
```

Comment:
You will want the filename to specify the full path otherwise it may not write the file where you expect. See example snippet from higher up in the function
```python
filename = os.path.expanduser(os.path.join(temp_dir, filename))
```

Also if this is actually running in production you will want this file to be uploaded to s3 bucket after you write it out. See get_target_key and s3_upload. For get_target_key, you will want to pass in a modified form of the naming_convention config. 

Example: I currently have our naming convention set to `meltano/{load_schema}/{stream}/{timestamp}.csv`. Would want schema file saved to `meltano/{load_schema}/{stream}/schema.json`. But logic should ideally be flexible. Alternate example `meltano/{load_schema}/{stream}-{timestamp}.csv` would save schema to `meltano/{load_schema}/schema.json`. 

Although that latter option would be dumb because it would keep overwriting schema files ontop of each other. 

Maybe we add optional new config `schema_naming_convention`. Then you could just set it how you want e.g. `meltano/{load_schema}/{stream}/schema.json` or `meltano/{load_schema}/{stream}-schema.json`

Reply:
I will update the path to include the temp_dir join. 

On the S3 piece I was assuming the for loop oustide that copies the files to s3 would handle since it is serialized it should get caught